### PR TITLE
tests/kernel/workq/work: Disable PM for ITE IT8XXX2

### PR DESCRIPTION
**Pull request by edersondisouza**
The workqueue test may intermittently fail, with errors such as:

  work_1cpu_test_1cpu_system_schedule: (elapsed_ms <= max_ms is false)
  long 102 > 101

Disabling PM makes the issues go away, and it is what this patch does.

### CODE REVIEW

**Review by ceolin — DISMISSED** [review]
That makes sense to me.

**Review by nashif — CHANGES_REQUESTED** [review]
The fix is to disable PM here boards/riscv/it82xx2_evb/it82xx2_evb_defconfig, we cant afford to have PM being disabled for each board individually in every test.

**Comment by edersondisouza** [comment]
> The fix is to disable PM here boards/riscv/it82xx2_evb/it82xx2_evb_defconfig, we cant afford to have PM being disabled for each board individually in every test.

Isn't this basically assuming PM doesn't work for the board? Is this right?

**Comment by nashif** [comment]
> Isn't this basically assuming PM doesn't work for the board? Is this right?

no, power management is not expected to be enabled by the platform definition. It is an application decision.

File: boards/riscv/it8xxx2_evb/it8xxx2_evb_defconfig [hunk 1]
```diff
@@ -19,7 +19,3 @@ CONFIG_INIT_STACKS=y
 CONFIG_LINKER_ORPHAN_SECTION_PLACE=y
 CONFIG_COMPILER_OPT="-mcmodel=medlow"
 CONFIG_GPIO=y
-# Power Management
-CONFIG_PM=y
-CONFIG_PM_DEVICE=y
-CONFIG_PM_POLICY_CUSTOM=y
```
Comment (nashif) [inline boards/riscv/it8xxx2_evb/it8xxx2_evb_defconfig:25]: given the custom policy, maybe add this:

```
--- a/boards/riscv/it8xxx2_evb/Kconfig.defconfig
+++ b/boards/riscv/it8xxx2_evb/Kconfig.defconfig
@@ -5,4 +5,13 @@ if BOARD_IT8XXX2_EVB

 config BOARD
        default "it8xxx2_evb"
+
+if PM
+config PM_DEVICE
+       default y
+
+config PM_POLICY_CUSTOM
+       default y
+endif
+
 endif
```

The we need to only enable PM in the app and everything else is set automatically.

**Comment by edersondisouza** [comment]
v2:
 - Enable `PM_DEVICE` and `PM_POLICY_CUSTOM` when PM is enabled by the app;
 - Rebased.

**Comment by edersondisouza** [comment]
Twister failures are unrelated, created a quick fix for it at https://github.com/zephyrproject-rtos/zephyr/pull/64667.

**Review by Dino-Li — APPROVED** [review]

**Comment by edersondisouza** [comment]
v3:
  Rebased, so that it has the fix for the unrelated CI failure.

**Review by nashif — APPROVED** [review]

**Pull request merged**